x-team/auto-ui
# Truncate long attached file names

**Release Type:** *Non-Breaking Feature* 


Fixes https://x-team-internal.atlassian.net/browse/XP-2493

## Description
> In order to improve user experience with files attachments, both on Notes and Portfolio questions, it's good to add some enhancements so that all the action buttons on the right are always visible and accessible. Let's truncate the attached file name so it does not overlap with action buttons next to it. Please use the same approach as we used for applicant names on applicant badges. For accessibility reasons, let's also show the full file name in a `title` attribute, so when we hover over the file we can see a full name in a tooltip.




## Checklist

**Before submitting a pull request,** please make sure the following is done:



## Related PRs



*N/A*

## Steps to Test or Reproduce
1- Run `npm start`
2- Go to `AttachFiles` component and open an example with files name
3- Reduce the width of the browser.
4 The file name should be truncated and appear a title when you make a hover.



## Impacted Areas in Application
- `AttachFiles` component



## Screenshots

#### `AttachFiles` component in  `auto-ui`

![truncate_file_name](https://user-images.githubusercontent.com/5392202/50791720-0eda6200-1299-11e9-84c6-c8398f078743.gif)

####  `AttachFiles` component used it in `note` section
![truncate_file_name_in_note_section](https://user-images.githubusercontent.com/5392202/50791754-21ed3200-1299-11e9-8fc8-8b4ca2551629.gif)
